examples/tina-cloud-starter-experimental/pages/_app.js

```diff
@@ -12,77 +12,37 @@ limitations under the License.
 */
 
 import dynamic from 'next/dynamic'
-import Link from 'next/link'
+import { TinaEditProvider } from 'tina-graphql-gateway/dist/edit-state'
 
-import { EditProvider, useEditState } from 'tina-graphql-gateway'
-
-// InnerApp that handles rendering edit mode or not
-function InnerApp({ Component, pageProps }) {
-  const { edit } = useEditState()
-  if (edit) {
-    // Dynamically load Tina only when in edit mode so it does not affect production
-    // see https://nextjs.org/docs/advanced-features/dynamic-import#basic-usage
-    const TinaWrapper = dynamic(() => import('../components/tina-wrapper'))
-    return (
-      <>
-        <TinaWrapper {...pageProps}>
-          {(props) => <Component {...props} />}
-        </TinaWrapper>
-        <EditToggle isInEditMode={true} />
-      </>
-    )
-  }
-  return (
-    <>
-      <Component {...pageProps} />
-      <EditToggle isInEditMode={true} />
-    </>
-  )
-}
-
-const EditToggle = (isInEditMode) => {
-  const { edit, setEdit } = useEditState()
-  return (
-    <>
-      {(Number(process.env.NEXT_PUBLIC_SHOW_EDIT_BTN) || edit) && (
-        <>
-          <button
-            onClick={() => {
-              setEdit(!edit)
-            }}
-            className="editLink"
-          >
-            {edit ? 'Exit edit mode' : 'Enter edit mode'}
-          </button>
-          <style jsx>{`
-            .editLink {
-              border: none;
-              position: fixed;
-              top: 0;
-              right: 0;
-              background: var(--orange);
-              color: var(--white);
-              padding: 0.5rem 0.75rem;
-              font-weight: bold;
-              text-decoration: none;
-              display: inline-block;
-              border-bottom-left-radius: 0.5rem;
-              cursor: pointer;
-              font-size: 20px;
-            }
-          `}</style>
-        </>
-      )}
-    </>
-  )
-}
+const TinaCMSProvider = dynamic(() => import('tina-graphql-gateway'), {
+  ssr: false,
+})
 
 // Our app is wrapped with edit provider
-function App(props) {
+function App({ Component, pageProps }) {
   return (
-    <EditProvider>
-      <InnerApp {...props} />
-    </EditProvider>
+    <TinaEditProvider
+      /** A button to toggle edit mode, recommended for demonstration purposes only */
+      showEditButton={true}
+      editMode={
+        <TinaCMSProvider
+          useUnstable={true}
+          config={{
+            branch: 'main',
+            clientId: process.env.NEXT_PUBLIC_TINA_CLIENT_ID,
+            organization: process.env.NEXT_PUBLIC_ORGANIZATION_NAME,
+            isLocalClient: Boolean(
+              Number(process.env.NEXT_PUBLIC_USE_LOCAL_CLIENT)
+            ),
+          }}
+          {...pageProps}
+        >
+          {(livePageProps) => <Component {...livePageProps} />}
```

Comment:
Would we have access to the CMS in here with `useCMS` ?

Reply:
Yeah, I'm not sure where is best, right now you can do this:
```tsx
<TinaCMSProvider
  useUnstable={true}
  cms={(cms) => cms.doSomething}
  // and formify
  formify={(args) =>doThing}
  config={{
    branch: 'main',
    clientId: process.env.NEXT_PUBLIC_TINA_CLIENT_ID,
    organization: process.env.NEXT_PUBLIC_ORGANIZATION_NAME,
    isLocalClient: Boolean(
        Number(process.env.NEXT_PUBLIC_USE_LOCAL_CLIENT)
      ),
    }
  }
  {...pageProps}
>
```

Reply:
I think we are already using the keyword `cms` for passing a CMS object.